server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioning.java

```diff
@@ -271,6 +274,9 @@ public ScimUser update(final String id, final ScimUser user, String zoneId) thro
         logger.debug("Updating user " + user.getUserName());
         final String origin = hasText(user.getOrigin()) ? user.getOrigin() : OriginKeys.UAA;
         user.setOrigin(origin);
+        if (isCheckOriginEnabled(zoneId)) {
+            checkOrigin(origin, zoneId);
```

Comment:
sonar does not show coverage for this here, can you please add a test and I would prefere an IT for negative testing as well

Reply:
Added test cannotUpdateUserWithInvalidOrigin